app/repository/MaillotRepositoryImpl.java

```diff
@@ -23,44 +24,65 @@ public MaillotRepositoryImpl(JPAApi jpaApi, DatabaseExecutionContext databaseExe
     }
 
     @Override
-    public CompletionStage<Stream<Maillot>> getAllMaillots() {
-        return supplyAsync(() -> wrap (this::getAllMaillots), databaseExecutionContext);
+    public CompletionStage<Stream<Maillot>> getAllMaillots(long stageId) {
+        return supplyAsync(() -> wrap (entityManager -> getAllMaillots(entityManager, stageId)), databaseExecutionContext);
     }
 
-    private Stream<Maillot> getAllMaillots(EntityManager em){
-        List<Maillot> maillots = em.createQuery("select m from Maillot m", Maillot.class).getResultList();
-        return maillots.stream();
+    private Stream<Maillot> getAllMaillots(EntityManager em, long stageId){
+        TypedQuery<Maillot> query = em.createQuery("select m from Maillot m where m.stage.id = :stageId" , Maillot.class);
+        query.setParameter("stageId", stageId);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "stageId" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)